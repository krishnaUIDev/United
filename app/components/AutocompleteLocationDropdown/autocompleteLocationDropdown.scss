@import "../../styles/tools";

.locationContainer {
  &:focus {
    outline-color: $modern-gold;
  }
  background: white;
  z-index: 4;
  @media screen and (min-width: $break-point-c) {
    position: absolute;
    width: 344px;
    height: auto;
    max-height: 376px;
    border: 1px solid $black-60;
    box-shadow: $box-shadow;
    border-radius: 3px;
  }
}

.locationListCloseBtn {
  position: absolute;
  top: 24px;
  right: 24px;
  @media screen and (min-width: $break-point-c) {
    right: 12px;
    top: 14px;
  }
}

.inputText {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.ariaAlertStyle {
  color: transparent;
  height: 1px;
  width: 1px;
  font-size: 1px;
}

.srOnly {
  @include visually-hidden;
  text-transform: none;
  font-size: 0;
  line-height: 0;
  height: 0;
}

.inputFieldContainer {
  margin-bottom: 0;
}

.locationListDiv {
  padding-top: 8px;
  padding-bottom: 24px;
  @media screen and (min-width: $break-point-c) {
    width: 99.9%; // cant be 100% or else will cut off border color
    padding-left: 0px;
    padding-top: 32px;
    padding-bottom: 8px;
  }
}
