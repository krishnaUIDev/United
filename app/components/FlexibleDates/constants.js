export const FLEXIBLE_MONTH = 'unitedapp/App/FLEXIBLE_MONTH';
export const FLEXIBLE_DAY = 'unitedapp/App/FLEXIBLE_DAY';
export const FLEXIBLE_DATE_DAY_OPTIONS = [
  {
    value: '0',
    label: '0 days',
  },
  {
    value: '1',
    label: '1 day',
  },
  {
    value: '2',
    label: '2 days',
  },
  {
    value: '3',
    label: '3 days',
  },
  {
    value: '4',
    label: '4 days',
  },
  {
    value: '5',
    label: '5 days',
  },
  {
    value: '6',
    label: '6 days',
  },
  {
    value: '7',
    label: '7 days',
  },
  {
    value: '8',
    label: '8 days',
  },
  {
    value: '9',
    label: '9 days',
  },
  {
    value: '10',
    label: '10 days',
  },
  {
    value: '11',
    label: '11 days',
  },
  {
    value: '12',
    label: '12 days',
  },
  {
    value: '13',
    label: '13 days',
  },
  {
    value: '14',
    label: '14 days',
  },
  {
    value: '15',
    label: '15 days',
  },
  {
    value: '16',
    label: '16 days',
  },
  {
    value: '17',
    label: '17 days',
  },
  {
    value: '18',
    label: '18 days',
  },
  {
    value: '19',
    label: '19 days',
  },
  {
    value: '20',
    label: '20 days',
  },
  {
    value: '21',
    label: '21 days',
  },
  {
    value: '22',
    label: '22 days',
  },
  {
    value: '23',
    label: '23 days',
  },
  {
    value: '24',
    label: '24 days',
  },
  {
    value: '25',
    label: '25 days',
  },
  {
    value: '30',
    label: '30 days',
  },
  {
    value: '60',
    label: '60 days',
  },
  {
    value: '90',
    label: '90 days',
  },
  {
    value: '120',
    label: '120 days',
  },
  {
    value: '180',
    label: '180 days',
  },
];
