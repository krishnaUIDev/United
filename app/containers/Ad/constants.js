export const API_ADS_REQUEST_URL = '/api/home/ads';

export const LOAD_ADS = 'unitedapp/App/LOAD_ADS';
export const LOAD_ADS_SUCCESS = 'unitedapp/App/LOAD_ADS_SUCCESS';
export const LOAD_ADS_ERROR = 'unitedapp/App/LOAD_ADS_ERROR';

/* LOCATIONS */
// TODO: Update these values with the correct string codes
export const LOCATION_CAROUSEL = 'BTWI';
export const LOCATION_LOGIN = 'HPIT';
export const LOCATION_SIDEBAR = 'SDBR';

