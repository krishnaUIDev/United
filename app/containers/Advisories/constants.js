
export const SEVERE_TYPE = 'severe';
export const WARNING_TYPE = 'warning';
