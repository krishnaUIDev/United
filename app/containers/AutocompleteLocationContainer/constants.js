/*
 * AutocompleteLocationContainer constants
 */

export const HIDE_AUTOCOMPLETE_LOCATION_DROPDOWN = 'unitedapp/App/HIDE_AUTOCOMPLETE_LOCATION_DROPDOWN';
export const UPDATE_MOBILE_VIEW = 'unitedapp/App/UPDATE_MOBILE_VIEW';
export const SET_AUTO_ARIA_LABEL = 'unitedapp/App/SET_AUTO_ARIA_LABEL';
