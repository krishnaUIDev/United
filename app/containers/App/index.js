/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';

import GlobalHeaderContainer from 'containers/GlobalHeaderContainer';
import GlobalFooterContainer from 'containers/GlobalFooterContainer';
import withProgressBar from 'components/ProgressBar';
import TimeoutModal from 'containers/TimeoutModal';
import Advisories from 'containers/Advisories';
// Reset stylesheets
import 'normalize.css/normalize.css';
import 'sanitize.css/sanitize.css';
import 'styles/styles.scss';
import config from 'config'; // eslint-disable-line

export function App(props) {
  return (
    <div className="page">
      <Helmet>
        <script src={config.ENSIGHTEN_BOOTSTRAP}></script>
      </Helmet>
      <Advisories />
      <a href="#travelTab" className="visuallyHidden">Skip to main content</a>
      <header role="banner">
        <GlobalHeaderContainer />
      </header>
      <TimeoutModal />
      {React.Children.toArray(props.children)}
      <footer role="contentinfo">
        <GlobalFooterContainer />
      </footer>
    </div>
  );
}

App.propTypes = {
  children: PropTypes.node,
};

export default withProgressBar(App);
