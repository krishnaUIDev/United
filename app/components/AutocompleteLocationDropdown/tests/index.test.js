import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import Immutable from 'immutable';
import { UP_ARROW_KEY, DOWN_ARROW_KEY } from 'containers/App/constants';
import AutocompleteLocationDropdown from '../index';

describe('<AutocompleteLocationDropdown />', () => {
  const arrowDownSpy = spy();
  const inputSpy = spy();
  const hideDropdownSpy = spy();
  const selectBookHotelSpy = spy();
  const mobileViewSpy = spy();
  const itemSelectedSpy = spy();
  const loadAirportSpy = jest.fn((value) => value);
  const setAriaSpy = spy();
  const dropdownListSpy = spy();
  const focusModelSpy = spy();
  const preventDefaultSpy = spy();
  const resetLocationCodeSpy = spy();
  const resetDisplayTextSpy = spy();
  const getRequestDataSpy = spy();
  const hideAutoLocation = Immutable.Map({
    bookCarOrigin: 'hidden',
    bookCarDropoff: 'hidden',
    bookHotel: 'visible',
    bookFlightOrigin: 'hidden',
    bookFlightDestination: 'hidden',
  });
  const mobileView = Immutable.Map({
    bookCarOrigin: 'relative',
    bookCarDropoff: 'relative',
    bookHotel: 'relative',
    bookFlightOrigin: 'relative',
    bookFlightDestination: 'relative',
  });
  const airportsList = ['YYZ', 'YKK', 'HFX', 'TOR', 'FRA'];
  let renderedComponent = shallow(
    <AutocompleteLocationDropdown
      hideAutocompleteLocation={hideAutoLocation}
      onArrowKeyDown={arrowDownSpy}
      hideAutocompleteLocationDropdown={hideDropdownSpy}
      onSelectInputItem={selectBookHotelSpy}
      updateMobileView={mobileViewSpy}
      mobileView={mobileView}
      onLoadAirports={loadAirportSpy}
      inputFieldValue={'foo'}
      inputFieldAriaLabel={'ariaLabel'}
      closePanelAriaLabel={'ariaLabel'}
      inputId="bookHotelInput"
      locationType="bookHotel"
      mobilePaddingLeft="8px"
      modelName="foo"
      inputType="text"
      inputFieldPlaceholder="placeholder"
      isRequired="true"
      isAriaRequired="true"
      onSetAutolocationAriaLabel={setAriaSpy}
      dropdownListFocus={dropdownListSpy}
      dataToPopulate={[{ foo: 'bar' }]}
      onFocusModel={focusModelSpy}
      locationCode="LAX"
      resetLocationCode={resetLocationCodeSpy}
      resetDisplayText={resetDisplayTextSpy}
      getRequestData={getRequestDataSpy}
      useImageIcons="true"
      listItemId="fooItem"
    />
  );
  const wrapper = renderedComponent.instance();
  beforeEach(() => {
    arrowDownSpy.reset();
    inputSpy.reset();
    hideDropdownSpy.reset();
    selectBookHotelSpy.reset();
    mobileViewSpy.reset();
    itemSelectedSpy.reset();
    // loadAirportSpy.reset();
    setAriaSpy.reset();
    dropdownListSpy.reset();
    focusModelSpy.reset();
    preventDefaultSpy.reset();
    resetLocationCodeSpy.reset();
    resetDisplayTextSpy.reset();
    getRequestDataSpy.reset();
  });
  it('render AutocompleteLocationDropdown Component', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
  it('render component prop locationCode', () => {
    expect(wrapper.props.locationCode).toEqual('LAX');
  });
  it('should load Airport', () => {
    wrapper.props.onLoadAirports(airportsList);
    expect(loadAirportSpy).toBeCalledWith(airportsList);
  });

  it('should render divs', () => {
    expect(renderedComponent.find('div').length).toEqual(6);
  });
  it('should render buttons', () => {
    expect(renderedComponent.find('#bookHotelInput').length).toEqual(1);
  });
  it('should render img', () => {
    expect(renderedComponent.find('img').length).toEqual(1);
  });
  it('should handle input keydown', () => {
    const input = renderedComponent.find('#bookHotelInput');
    input.simulate('keydown');
    expect(arrowDownSpy.called).toEqual(true);
  });
  it('should handle onPickupLocationChange()', () => {
    expect(renderedComponent.instance().onPickupLocationChange({ target: { value: 'ch' } })).toEqual(false);
  });
  it('should render getDropdownItem()', () => {
    const dataArray = [{
      firstLine: 'Limburg',
      secondLine: 'Germany',
      id: '3',
    }];
    const item = renderedComponent.instance().getDropdownItem(dataArray)[0].props;
    expect(item.firstLineText).toEqual('Limburg');
    expect(item.topLineId).toEqual('topLine3');
    expect(item.locationType).toEqual('bookHotel');
    expect(item.bottomLineId).toEqual('bottomLine3');
    expect(item.bottomLineText).toEqual('Germany');
  });
  it('should handle Control.text onFocus', () => {
    const component = renderedComponent.find('#bookHotelInput');
    component.simulate('focus', { target: { value: 'foo' } });
    expect(inputSpy.called).toEqual(false);
  });
  it('should handle close panel button click', () => {
    const btn = renderedComponent.find('#locationListCloseBtn');
    btn.simulate('click');
    setTimeout(() => {
      expect(hideDropdownSpy.called).toEqual(true);
    }, 1);
  });
  it('should handle close panel button blur', () => {
    const btn = renderedComponent.find('#locationListCloseBtn');
    btn.simulate('focus');
    btn.simulate('blur');
    setTimeout(() => {
      expect(hideDropdownSpy.called).toEqual(true);
    }, 1);
  });
  it('should handle close panel button down arrow key down', () => {
    const btn = renderedComponent.find('#locationListCloseBtn');
    btn.simulate('keydown', { preventDefault: preventDefaultSpy, keyCode: DOWN_ARROW_KEY, target: { id: 'locationListCloseBtn' } });
    expect(preventDefaultSpy.called).toEqual(true);
    expect(focusModelSpy.called).toEqual(true);
  });
  it('should handle close panel button down arrow key up', () => {
    const btn = renderedComponent.find('#locationListCloseBtn');
    btn.simulate('keydown', { preventDefault: preventDefaultSpy, keyCode: UP_ARROW_KEY, target: { id: 'locationListCloseBtn' } });
    expect(preventDefaultSpy.called).toEqual(true);
    expect(focusModelSpy.called).toEqual(true);
  });
  it('should handle getAriaLabel()', () => {
    renderedComponent.instance().getAriaLabel();
    expect(setAriaSpy.called).toEqual(true);
  });
  it('should return true for onCheckAcceptedActiveElements() with the param listItem', () => {
    expect(renderedComponent.instance().onCheckAcceptedActiveElements('listItem'));
  });
  it('should return false for onCheckAcceptedActiveElements() with the param foo', () => {
    expect(renderedComponent.instance().onCheckAcceptedActiveElements('foo'));
  });
  it('should handle close button blur', () => {
    renderedComponent.find('#locationListCloseBtn').simulate('blur');
    setTimeout(() => {
      expect(hideDropdownSpy.called).toEqual(true);
    }, 1);
  });
  it('should handle input field blur', () => {
    renderedComponent.find('#bookHotelInput').simulate('blur');
  });
  it('should handle input field onChange', () => {
    renderedComponent.find('#bookHotelInput').simulate('change', { target: { value: 'foo' } });
    expect(resetLocationCodeSpy.called).toEqual(true);
    expect(resetDisplayTextSpy.called).toEqual(true);
    expect(getRequestDataSpy.called).toEqual(true);
  });
  it('should handle getAriaLabel() with inputValue length less than 3', () => {
    renderedComponent.instance().getAriaLabel('lo');
    expect(setAriaSpy.called).toEqual(true);
  });
  it('should handle getAriaLabel() with inputValue length equal to 3 and visible dropdown', () => {
    renderedComponent.instance().getAriaLabel('lon');
    expect(setAriaSpy.called).toEqual(true);
  });
  it('should handle getAriaLabel() with inputValue length greater than 2 with dropdown hidden', () => {
    renderedComponent = shallow(
      <AutocompleteLocationDropdown
        hideAutocompleteLocation={hideAutoLocation}
        onArrowKeyDown={arrowDownSpy}
        hideAutocompleteLocationDropdown={hideDropdownSpy}
        onSelectInputItem={selectBookHotelSpy}
        updateMobileView={mobileViewSpy}
        mobileView={mobileView}
        onLoadAirports={loadAirportSpy}
        inputFieldValue={'foo'}
        inputFieldAriaLabel={'ariaLabel'}
        closePanelAriaLabel={'ariaLabel'}
        inputId="bookHotelInput"
        locationType="bookHotel"
        mobilePaddingLeft="8px"
        modelName="foo"
        inputType="text"
        inputFieldPlaceholder="placeholder"
        isRequired="true"
        isAriaRequired="true"
        onSetAutolocationAriaLabel={setAriaSpy}
        dropdownListFocus={dropdownListSpy}
        dataToPopulate={[{ foo: 'bar' }]}
        onFocusModel={focusModelSpy}
        locationCode="LAX"
        resetLocationCode={resetLocationCodeSpy}
        resetDisplayText={resetDisplayTextSpy}
        getRequestData={getRequestDataSpy}
      />
    );
    renderedComponent.instance().getAriaLabel('london');
    expect(setAriaSpy.called).toEqual(true);
  });
});
