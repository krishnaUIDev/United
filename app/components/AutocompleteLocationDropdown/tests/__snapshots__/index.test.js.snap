// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AutocompleteLocationDropdown /> render AutocompleteLocationDropdown Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        style={
            Object {
                "paddingBottom": "10px",
              }
        }
    >
        <div
            className=""
            style={
                Object {
                    "marginTop": "0",
                    "paddingBottom": "0",
                    "paddingLeft": "8px",
                  }
            }
        >
            <DefaultConnectedControlText
                aria-required="true"
                autoComplete="off"
                className=""
                data-required="true"
                id="bookHotelInput"
                model="foo"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="placeholder"
                spellCheck="false"
                tabIndex="0"
                type="text"
                validators={
                    Object {
                        "isValidInput": [Function],
                      }
                }
            />
            <label
                className={undefined}
                htmlFor="bookHotelInput"
            >
                placeholder
            </label>
            <div
                aria-live="polite"
                className={undefined}
            />
            <Modeled(Connect(Errors))
                messages={
                    Object {
                        "isValidInput": "undefined",
                      }
                }
                model="foo"
                show={[Function]}
                wrapper={[Function]}
            />
        </div>
        <div
            className={undefined}
            id="locationContainer"
            style={
                Object {
                    "display": "inline",
                    "height": "auto",
                    "visibility": "visible",
                  }
            }
        >
            <div
                className={undefined}
            >
                <AutocompleteLocationDropdownItem
                    ariaLocationTxt={undefined}
                    bottomLineId=""
                    bottomLineText=""
                    displayText=""
                    firstLineText={undefined}
                    fullModelName={undefined}
                    hideDropdown={[Function]}
                    imgSrc={null}
                    imgStyle={null}
                    itemId={undefined}
                    listLength={1}
                    locationCode=""
                    locationTxt={undefined}
                    locationType="bookHotel"
                    onItemClick={undefined}
                    onItemKeyDown={[Function]}
                    parentModel={undefined}
                    topLineId="topLineundefined"
                    useImageIcons="true"
                />
            </div>
            <DefaultConnectedControlButton
                aria-label="ariaLabel"
                className={undefined}
                id="locationListCloseBtn"
                model=".closeAutoList"
                onBlur={[Function]}
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex="0"
                title="ariaLabel"
            >
                <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="IMAGE_MOCK"
                />
            </DefaultConnectedControlButton>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            style={
                  Object {
                        "paddingBottom": "10px",
                      }
            }
      >
            <div
                  className=""
                  style={
                        Object {
                              "marginTop": "0",
                              "paddingBottom": "0",
                              "paddingLeft": "8px",
                            }
                  }
            >
                  <DefaultConnectedControlText
                        aria-required="true"
                        autoComplete="off"
                        className=""
                        data-required="true"
                        id="bookHotelInput"
                        model="foo"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="placeholder"
                        spellCheck="false"
                        tabIndex="0"
                        type="text"
                        validators={
                              Object {
                                    "isValidInput": [Function],
                                  }
                        }
                  />
                  <label
                        className={undefined}
                        htmlFor="bookHotelInput"
                  >
                        placeholder
                  </label>
                  <div
                        aria-live="polite"
                        className={undefined}
                  />
                  <Modeled(Connect(Errors))
                        messages={
                              Object {
                                    "isValidInput": "undefined",
                                  }
                        }
                        model="foo"
                        show={[Function]}
                        wrapper={[Function]}
                  />
            </div>
            <div
                  className={undefined}
                  id="locationContainer"
                  style={
                        Object {
                              "display": "inline",
                              "height": "auto",
                              "visibility": "visible",
                            }
                  }
            >
                  <div
                        className={undefined}
                  >
                        <AutocompleteLocationDropdownItem
                              ariaLocationTxt={undefined}
                              bottomLineId=""
                              bottomLineText=""
                              displayText=""
                              firstLineText={undefined}
                              fullModelName={undefined}
                              hideDropdown={[Function]}
                              imgSrc={null}
                              imgStyle={null}
                              itemId={undefined}
                              listLength={1}
                              locationCode=""
                              locationTxt={undefined}
                              locationType="bookHotel"
                              onItemClick={undefined}
                              onItemKeyDown={[Function]}
                              parentModel={undefined}
                              topLineId="topLineundefined"
                              useImageIcons="true"
                        />
                  </div>
                  <DefaultConnectedControlButton
                        aria-label="ariaLabel"
                        className={undefined}
                        id="locationListCloseBtn"
                        model=".closeAutoList"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex="0"
                        title="ariaLabel"
                  >
                        <img
                              alt=""
                              aria-hidden="true"
                              role="presentation"
                              src="IMAGE_MOCK"
                        />
                  </DefaultConnectedControlButton>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <AutocompleteLocationDropdown
        closePanelAriaLabel="ariaLabel"
        dataToPopulate={
                Array [
                        Object {
                          "foo": "bar",
                        },
                      ]
        }
        dropdownListFocus={[Function]}
        getRequestData={[Function]}
        hideAutocompleteLocation={
                Immutable.Map {
                        bookCarOrigin: "hidden",
                        bookCarDropoff: "hidden",
                        bookHotel: "visible",
                        bookFlightOrigin: "hidden",
                        bookFlightDestination: "hidden",
                }
        }
        hideAutocompleteLocationDropdown={[Function]}
        inputFieldAriaLabel="ariaLabel"
        inputFieldPlaceholder="placeholder"
        inputFieldValue="foo"
        inputId="bookHotelInput"
        inputType="text"
        isAriaRequired="true"
        isRequired="true"
        listItemId="fooItem"
        locationCode="LAX"
        locationType="bookHotel"
        mobilePaddingLeft="8px"
        mobileView={
                Immutable.Map {
                        bookCarOrigin: "relative",
                        bookCarDropoff: "relative",
                        bookHotel: "relative",
                        bookFlightOrigin: "relative",
                        bookFlightDestination: "relative",
                }
        }
        modelName="foo"
        onArrowKeyDown={[Function]}
        onFocusModel={[Function]}
        onLoadAirports={[Function]}
        onSelectInputItem={[Function]}
        onSetAutolocationAriaLabel={[Function]}
        resetDisplayText={[Function]}
        resetLocationCode={[Function]}
        updateMobileView={[Function]}
        useImageIcons="true"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": AutocompleteLocationDropdown {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "closePanelAriaLabel": "ariaLabel",
          "dataToPopulate": Array [
            Object {
              "foo": "bar",
            },
          ],
          "dropdownListFocus": [Function],
          "getRequestData": [Function],
          "hideAutocompleteLocation": Immutable.Map {
            bookCarOrigin: "hidden",
            bookCarDropoff: "hidden",
            bookHotel: "visible",
            bookFlightOrigin: "hidden",
            bookFlightDestination: "hidden",
},
          "hideAutocompleteLocationDropdown": [Function],
          "inputFieldAriaLabel": "ariaLabel",
          "inputFieldPlaceholder": "placeholder",
          "inputFieldValue": "foo",
          "inputId": "bookHotelInput",
          "inputType": "text",
          "isAriaRequired": "true",
          "isRequired": "true",
          "listItemId": "fooItem",
          "locationCode": "LAX",
          "locationType": "bookHotel",
          "mobilePaddingLeft": "8px",
          "mobileView": Immutable.Map {
            bookCarOrigin: "relative",
            bookCarDropoff: "relative",
            bookHotel: "relative",
            bookFlightOrigin: "relative",
            bookFlightDestination: "relative",
},
          "modelName": "foo",
          "onArrowKeyDown": [Function],
          "onFocusModel": [Function],
          "onLoadAirports": [Function],
          "onSelectInputItem": [Function],
          "onSetAutolocationAriaLabel": [Function],
          "resetDisplayText": [Function],
          "resetLocationCode": [Function],
          "updateMobileView": [Function],
          "useImageIcons": "true",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    style={
                              Object {
                                        "paddingBottom": "10px",
                                      }
                    }
          >
                    <div
                              className=""
                              style={
                                        Object {
                                                  "marginTop": "0",
                                                  "paddingBottom": "0",
                                                  "paddingLeft": "8px",
                                                }
                              }
                    >
                              <DefaultConnectedControlText
                                        aria-required="true"
                                        autoComplete="off"
                                        className=""
                                        data-required="true"
                                        id="bookHotelInput"
                                        model="foo"
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="placeholder"
                                        spellCheck="false"
                                        tabIndex="0"
                                        type="text"
                                        validators={
                                                  Object {
                                                            "isValidInput": [Function],
                                                          }
                                        }
                              />
                              <label
                                        className={undefined}
                                        htmlFor="bookHotelInput"
                              >
                                        placeholder
                              </label>
                              <div
                                        aria-live="polite"
                                        className={undefined}
                              />
                              <Modeled(Connect(Errors))
                                        messages={
                                                  Object {
                                                            "isValidInput": "undefined",
                                                          }
                                        }
                                        model="foo"
                                        show={[Function]}
                                        wrapper={[Function]}
                              />
                    </div>
                    <div
                              className={undefined}
                              id="locationContainer"
                              style={
                                        Object {
                                                  "display": "inline",
                                                  "height": "auto",
                                                  "visibility": "visible",
                                                }
                              }
                    >
                              <div
                                        className={undefined}
                              >
                                        <AutocompleteLocationDropdownItem
                                                  ariaLocationTxt={undefined}
                                                  bottomLineId=""
                                                  bottomLineText=""
                                                  displayText=""
                                                  firstLineText={undefined}
                                                  fullModelName={undefined}
                                                  hideDropdown={[Function]}
                                                  imgSrc={null}
                                                  imgStyle={null}
                                                  itemId={undefined}
                                                  listLength={1}
                                                  locationCode=""
                                                  locationTxt={undefined}
                                                  locationType="bookHotel"
                                                  onItemClick={undefined}
                                                  onItemKeyDown={[Function]}
                                                  parentModel={undefined}
                                                  topLineId="topLineundefined"
                                                  useImageIcons="true"
                                        />
                              </div>
                              <DefaultConnectedControlButton
                                        aria-label="ariaLabel"
                                        className={undefined}
                                        id="locationListCloseBtn"
                                        model=".closeAutoList"
                                        onBlur={[Function]}
                                        onClick={[Function]}
                                        onKeyDown={[Function]}
                                        role="button"
                                        tabIndex="0"
                                        title="ariaLabel"
                              >
                                        <img
                                                  alt=""
                                                  aria-hidden="true"
                                                  role="presentation"
                                                  src="IMAGE_MOCK"
                                        />
                              </DefaultConnectedControlButton>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    style={
                              Object {
                                        "paddingBottom": "10px",
                                      }
                    }
          >
                    <div
                              className=""
                              style={
                                        Object {
                                                  "marginTop": "0",
                                                  "paddingBottom": "0",
                                                  "paddingLeft": "8px",
                                                }
                              }
                    >
                              <DefaultConnectedControlText
                                        aria-required="true"
                                        autoComplete="off"
                                        className=""
                                        data-required="true"
                                        id="bookHotelInput"
                                        model="foo"
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        onKeyDown={[Function]}
                                        placeholder="placeholder"
                                        spellCheck="false"
                                        tabIndex="0"
                                        type="text"
                                        validators={
                                                  Object {
                                                            "isValidInput": [Function],
                                                          }
                                        }
                              />
                              <label
                                        className={undefined}
                                        htmlFor="bookHotelInput"
                              >
                                        placeholder
                              </label>
                              <div
                                        aria-live="polite"
                                        className={undefined}
                              />
                              <Modeled(Connect(Errors))
                                        messages={
                                                  Object {
                                                            "isValidInput": "undefined",
                                                          }
                                        }
                                        model="foo"
                                        show={[Function]}
                                        wrapper={[Function]}
                              />
                    </div>
                    <div
                              className={undefined}
                              id="locationContainer"
                              style={
                                        Object {
                                                  "display": "inline",
                                                  "height": "auto",
                                                  "visibility": "visible",
                                                }
                              }
                    >
                              <div
                                        className={undefined}
                              >
                                        <AutocompleteLocationDropdownItem
                                                  ariaLocationTxt={undefined}
                                                  bottomLineId=""
                                                  bottomLineText=""
                                                  displayText=""
                                                  firstLineText={undefined}
                                                  fullModelName={undefined}
                                                  hideDropdown={[Function]}
                                                  imgSrc={null}
                                                  imgStyle={null}
                                                  itemId={undefined}
                                                  listLength={1}
                                                  locationCode=""
                                                  locationTxt={undefined}
                                                  locationType="bookHotel"
                                                  onItemClick={undefined}
                                                  onItemKeyDown={[Function]}
                                                  parentModel={undefined}
                                                  topLineId="topLineundefined"
                                                  useImageIcons="true"
                                        />
                              </div>
                              <DefaultConnectedControlButton
                                        aria-label="ariaLabel"
                                        className={undefined}
                                        id="locationListCloseBtn"
                                        model=".closeAutoList"
                                        onBlur={[Function]}
                                        onClick={[Function]}
                                        onKeyDown={[Function]}
                                        role="button"
                                        tabIndex="0"
                                        title="ariaLabel"
                              >
                                        <img
                                                  alt=""
                                                  aria-hidden="true"
                                                  role="presentation"
                                                  src="IMAGE_MOCK"
                                        />
                              </DefaultConnectedControlButton>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <AutocompleteLocationDropdown
    closePanelAriaLabel="ariaLabel"
    dataToPopulate={
        Array [
            Object {
              "foo": "bar",
            },
          ]
    }
    dropdownListFocus={[Function]}
    getRequestData={[Function]}
    hideAutocompleteLocation={
        Immutable.Map {
            bookCarOrigin: "hidden",
            bookCarDropoff: "hidden",
            bookHotel: "visible",
            bookFlightOrigin: "hidden",
            bookFlightDestination: "hidden",
        }
    }
    hideAutocompleteLocationDropdown={[Function]}
    inputFieldAriaLabel="ariaLabel"
    inputFieldPlaceholder="placeholder"
    inputFieldValue="foo"
    inputId="bookHotelInput"
    inputType="text"
    isAriaRequired="true"
    isRequired="true"
    listItemId="fooItem"
    locationCode="LAX"
    locationType="bookHotel"
    mobilePaddingLeft="8px"
    mobileView={
        Immutable.Map {
            bookCarOrigin: "relative",
            bookCarDropoff: "relative",
            bookHotel: "relative",
            bookFlightOrigin: "relative",
            bookFlightDestination: "relative",
        }
    }
    modelName="foo"
    onArrowKeyDown={[Function]}
    onFocusModel={[Function]}
    onLoadAirports={[Function]}
    onSelectInputItem={[Function]}
    onSetAutolocationAriaLabel={[Function]}
    resetDisplayText={[Function]}
    resetLocationCode={[Function]}
    updateMobileView={[Function]}
    useImageIcons="true"
/>,
}
`;
